src/update.c

```diff
@@ -365,6 +367,37 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Get a list of all the iterative manifests from the server_manifest that match
+	* the "from" version, also get a list of all the full manifests in the MoM */
+	struct list *list1;
+	struct file *file1;
+
+	list1 = server_manifest->manifests;
+	while (list1) {
+		file1 = list1->data;
+		list1 = list1->next;
+		/* Keep only iterative manifests that are from the current version */
+		if (file1->is_iterative == 1 && file1->from_version == current_version) {
+			iterative_manifests = list_prepend_data(iterative_manifests, file1);
+		} else {
+			full_manifests = list_prepend_data(full_manifests, file1);
+		}
+	}
+
+	/* Add the full manifest of those bundles that do not have a valid iterative manifest
+	* into the list of iterative_manifests */
+	list1 = full_manifests;
+	while (list1) {
+		file1 = list1->data;
+		list1 = list1->next;
+		if (!search_bundle_in_manifests(iterative_manifests, file1->bundlename)) {
+			iterative_manifests = list_prepend_data(iterative_manifests, file1);
+		}
+	}
+
+	/* Replace the server manifests with the list that includes the iterative manifests */
+	server_manifest->manifests = iterative_manifests;
```

Comment:
this will leak data. you need to free the list nodes from server_manifest->manifests.